### ISSUE TITLE
All pods will restart when enable feature-gates InPlacePodVerticalScaling of apiserver.

### ISSUE THREAD

**Issue by payall4u** [kubernetes/kubernetes#126328]
### What happened?

All pods will restart when enable feature-gates InPlacePodVerticalScaling of apiserver. I'm not sure if this is a bug or it should be an enhancement.

### What did you expect to happen?

I hope I can enable InPlacePodVerticalScaling in some existing clusters without restarting pods, at least not on a large scale.

### How can we reproduce it (as minimally and precisely as possible)?

Create some pods in the cluster, and enable feature-gates InPlacePodVerticalScaling of apiserver. 

Then all pod will restart💥💥💥.

### Anything else we need to know?

There is no ResizePolicy field when kubelet calculates the hash value of the existing pod. 

And apiserver will add the default value of spec.containers.ResizePolicy to the pod after turning on InPlacePodVerticalScaling. 

Then kubelet will sean the spec.containers.ResizePolicy and found hash changed.

### Kubernetes version

```console
$ kubectl version
Client Version: v1.28.3
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.3
```

### Cloud provider

NOP

**Comment by payall4u** [comment]
I am not sure whether I should adapt it on kubelet or apiserver. 
/sig node

**Comment by payall4u** [comment]
Need help. @vinaykul @SergeyKanzhelev

**Comment by HirazawaUi** [comment]
This has been fixed in v1.31 #124220, And because this is a beta feature, we will not cherry-pick it to older versions.

**Comment by HirazawaUi** [comment]
If you think this issue still needs to be discussed, please feel free to open
/close
